### DISCUSSION TITLE
broadening scope of TikZImage.pm

### DISCUSSION THREAD

**Discussion by Alex-Jordan** [openwebwork/pg#605] [Ideas]
It significantly complicates PreTeXt-WeBWorK relations to have TikZImage.pm assume the `\begin{tikzpicture}...\end{tikzpicture}` wrapper. I would like to do one of the following:

1. Add a flag that suppresses the automatic `\begin{tikzpicture}...\end{tikzpicture}` wrapper. So if this flag is used, the incoming tex code could include that wrapper directly. The incoming tex code could even skip tikz entirely and (for example) make an xypic image.
2. Make something parallel to TikZImage.pm that might be called LaTeXImage.pm. For more general purpose creation of images through LaTeX.
3. Like item 2, but LaTeXImage.pm would replace TikZImage.pm. And then PGtikz.pl would change to incorporate the tikz-specific elements before passing code to LaTeXImage.pm.

Does the overarching goal raise any red flags? If not, is there a preference for one of these approaches?

**Comment by drgrice1** [comment]
I don't like 1.  The library and macro have the name tikz in them, so it seems wrong for it to be used in a more general manner.  So 2 or 3 would be better.  I would recommend 2 initially, and if that is done in such a way that 3 is possible, then move to 3.

**Comment by Alex-Jordan** [comment]
Since I posted this morning, I've been working toward option 3. Almost ready for a PR to develop, probably within the hour.

**Comment by Alex-Jordan** [comment]
#606 addressed this.
